### ISSUE TITLE
Provide more helpful error message when trying to use const generic defaults

### ISSUE THREAD

**Issue by XAMPPRocky** [rust-lang/rust#80507]
Defaults are currently not implemented in const generics. With `min_const_generics` stabilising soon, I've been using them more and one thing that is surprising is the lack of defaults. Especially when you try to declare a default, the error message provides a unexpected syntax error, when I would expect a "defaults in const generics is not implemented" error, leading you to think that there's another syntax that you need to use as opposed to it just not being available.

```rust
#![feature(min_const_generics)]

fn foo<const NUM: usize = 0>() -> usize {
    NUM
}
```

### Error Message
```
error: expected one of `!`, `(`, `+`, `,`, `::`, `<`, or `>`, found `=`
 --> src/lib.rs:3:25
  |
3 | fn foo<const NUM: usize = 0>() -> usize {
  |                         ^ expected one of 7 possible tokens

error: aborting due to previous error
```

**Comment by varkor** [comment]
Perhaps the simplest solution is to extract the parsing for const generic defaults from https://github.com/rust-lang/rust/pull/75384, but unconditionally emit an error in `ast_validation.rs` if they're used. I think it's very likely we'll want to support const generic defaults before too long, so hopefully the solution is going to be temporary anyway.

cc @JulianKnodt in case you're interested fixing this.
